### DISCUSSION TITLE
Reminders are being removed during sync operations

### DISCUSSION THREAD

**Discussion by Billbarasch** [phw198/OutlookGoogleCalendarSync#1980] [Q&A]
Hi, I do bidirectional sync between outlook and google calendar. My configuration is to sync
location, Description, Attendees and
Reminders: I find that reminders are being removed 

I see in the console  Reminder: 10 => **removed** for no apparent reason

There should be no reason for the reminder to be removed. This can cause me to miss meetings because of lack missing reminder. How can I prevent the reminder from being removed. I do want the reminder to by synced in both directions.

```
2024-09-26 14:26:14,316 ULTRA-FINE 3  OutlookGoogleCalendarSync.Sync.Engine [231] -  Outlook attribute: opaque
2024-09-26 14:26:14,356 DEBUG 3  OutlookGoogleCalendarSync.Console [364] -  Reminder: 10 => removed
2024-09-26 14:26:14,356 ULTRA-FINE 1  OutlookGoogleCalendarSync.Console [209] -  Console navigating.
2024-09-26 14:26:14,357 DEBUG 3  OutlookGoogleCalendarSync.Console [364] -  1 attributes updated.
2024-09-26 14:26:14,362 ULTRA-FINE 1  OutlookGoogleCalendarSync.Console [209] -  Console navigating.
2024-09-26 14:26:14,367 ULTRA-FINE 1  OutlookGoogleCalendarSync.Console [218] -  Navigated status = Loading
2024-09-26 14:26:14,372 ULTRA-FINE 1  OutlookGoogleCalendarSync.Console [218] -  Console finished navigating
2024-09-26 14:26:14,376 ULTRA-FINE 1  OutlookGoogleCalendarSync.Console [202] -  Document completed.
```

**Comment by phw198** [comment]
You say you are or you are not syncing reminders? It's not clear from your post with the line break.

If not syncing reminders, it would be a duplicate of https://github.com/phw198/OutlookGoogleCalendarSync/issues/1458#issuecomment-2171939952

Again, to triage further, you'd need to raise a bug report and provide your full log file capturing the reproduced issue.

**Reply by Billbarasch** [reply]
Hi PHW198, 
Thanks for picking up my question. 

What I am trying to describe is that in my original configuration my reminders were being removed by OGCS during sync. 
This is what I am seeing in the console. 
Reminder: 10 => removed

I do not  want the reminder to be removed. 
my current configuration is 
![OGCS-sync-config](https://github.com/user-attachments/assets/0ffa611f-a6b8-474e-bf4e-b5997d18ab4a)

I am willing to open a bug report if that is needed. 

I very much appreciate OGCS and your efforts.

**Reply by phw198** [reply]
So with the configuration set to not sync reminders, they were being removed - that's expected as per #1458.
Now you have set OGCS to sync reminders...and they are no longer being removed?
I'm struggling to understand what you are reporting that isn't already covered by #1458 😕

**Reply by Billbarasch** [reply]
Hi phw198. 
As you can see in the screen shot, I am trying to sync reminders. 
The first sync after adding the event the reminder is synced regardless if from google to outlook or from outlook to google. 
In subsequent syncs the reminder is removed  Reminder: 10 => removed. 
Again I do want to sync reminders and have reminders box synced.

**Reply by Billbarasch** [reply]
[OGcalsync.log](https://github.com/user-attachments/files/18338568/OGcalsync.log)

**Reply by phw198** [reply]
Hi @Billbarasch, I've finally got around to having a look at this - this is what I can tell from your log file:

1. The `1/8/2025 16:00 => "NFL Interview Prep "` item is already synced by OGCS between the two calendars.
2. The Google item is deleted and recreated by something outside of OGCS - this has no reminder
3. OGCS sync and "reclaims" the recreated event, thereby tying it back to the Outlook counterpart
    1. In the same sync, OGCS then assigns the 15 minutes reminder to Google
5.  Push Sync detects a change in Outlook and triggers another 2 way sync 2 minutes later
6. Outlook's item is newer than when OGCS just updated Google.
7. The organiser of the Outlook item is apparently not you and they have altered the invite (as proven by a change in the IDs)
8. OGCS compares Outlook to Google and determines Outlook no longer has a reminder and removes it from Google.

So....from this I surmise that the organiser of this event removed the reminder from your Outlook invite and OGCS is simply syncing that. I also am going to guess that the organiser also invited your Gmail account, which is why that was recreated, but OGCS synced and reclaimed it before the updated invite was seen by Outlook - that came in within the next 2 minutes.

I certainly can't replicate OGCS randomly removing reminders that exist in the most recently updated calendar event.
